### PR TITLE
Add magnitude and function name interfaces. Refs #19555

### LINKED ISSUE #19555: Extract magnitude and function from IntegralPreservingFunctionIC
## Reason
When adding the `IntegralPreservingFunctionIC` in #19476, I forgot to add a few interface functions needed for Cardinal to automatically set up ICs using Actions.

## Design
Add functions that return the function name to `FunctionIC` and the magnitude to `IntegralPreservingFunctionIC`.

## Impact
Additional API, no expected changes.

### PR DESCRIPTION
**Pull request by aprilnovak**
Adds necessary APIs for Cardinal to set up these ICs automatically, with a subsequent IC that depends on the function provided to the `IntegralPreservingFunctionIC`.

Closes #19555

### CODE REVIEW

File: framework/include/ics/FunctionIC.h [hunk 1]
```diff
@@ -35,6 +35,12 @@ class FunctionIC : public InitialCondition
 
   FunctionIC(const InputParameters & parameters);
 
+  /**
+   * Get the function name
+   * @return function name
+   */
+  const FunctionName & functionName() const { return _function_name; }
```
Comment (loganharbour) [inline framework/include/ics/FunctionIC.h:42]: No need to store `_function_name`, remove that member and just do

```
const FunctionName &
FunctionIC::functionName() const
{
  return _func.name();
}
```

in source

**Review by loganharbour — CHANGES_REQUESTED** [review]

**Review by loganharbour — COMMENTED** [review]
You need to do it in the source.

`Function` isn't included in the header, it's just forward declared

**Comment by aprilnovak** [comment]
> You need to do it in the source.

Will it fail a non-unity build, you mean? Because it does compile as-in in the header

**Comment by loganharbour** [comment]
> > You need to do it in the source.
> 
> Will it fail a non-unity build, you mean? Because it does compile as-in in the header

Correct.

File: framework/include/ics/FunctionIC.h [hunk 2]
```diff
@@ -35,6 +35,12 @@ class FunctionIC : public InitialCondition
 
   FunctionIC(const InputParameters & parameters);
 
+  /**
+   * Get the function name
+   * @return function name
```
Comment (loganharbour) [inline framework/include/ics/FunctionIC.h:40]:
```suggestion
   * @returns The function name
```

**Review by loganharbour — CHANGES_REQUESTED** [review]
Sorry wasn't very thorough the first time around. Good after this

**Comment by aprilnovak** [comment]
Thanks for the quick review!

**Review by loganharbour — APPROVED** [review]

**Pull request merged**